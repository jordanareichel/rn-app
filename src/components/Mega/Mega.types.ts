export type MegaProps = {
  quantity: number;
  onChangeValue: (text: any) => void;
  onPress: () => void;
};
