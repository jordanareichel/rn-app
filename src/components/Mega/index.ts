export * from './Mega.comp';
export * from './Mega.types';
