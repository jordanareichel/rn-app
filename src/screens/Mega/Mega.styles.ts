import styled from 'styled-components/native';

export const Body = styled.View`
  flex: 1;
  padding: 32px 16px 0;
  justify-content: space-between;
`;

export const Row = styled.View`
  flex-direction: row;
  align-items: center;
  justify-content: center;
  margin-top: 22px;
  flex-wrap: wrap;
`;
