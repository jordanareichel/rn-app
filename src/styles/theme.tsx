export const theme = {
  font: {
    sm: '12',
    md: '14',
    lg: '18',
  },
  colors: {
    dark: '#000000',
    light: '#ffffff',
  },
};
