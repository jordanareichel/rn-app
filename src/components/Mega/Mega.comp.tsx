import {Input} from '@components/Input';
import {Text} from '@components/Text';
import React from 'react';
import {ButtonGenerate, Wrapper} from './Mega.styles';
import {MegaProps} from './Mega.types';

export const Mega: React.FC<MegaProps> = props => {
  const {quantity, onChangeValue, onPress} = props;

  return (
    <Wrapper>
      <Input
        placeholder="Digite um nÃºmero"
        value={String(quantity)}
        onChangeText={onChangeValue}
        keyboardType={'numeric'}
      />
      <ButtonGenerate onPress={onPress} disabled={!quantity}>
        <Text fontSize={16} color="white" fontWeight="bold">
          Gerar
        </Text>
      </ButtonGenerate>
    </Wrapper>
  );
};
