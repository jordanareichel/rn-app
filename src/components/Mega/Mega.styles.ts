import {theme} from '@styles/theme';
import _ from '@utils/object';
import styled from 'styled-components/native';

export const Wrapper = styled.View`
  flex: 1;
  margin-top: 32px;
`;

export const ButtonGenerate = styled.TouchableOpacity`
  padding: 16px;
  background-color: ${_.get(theme, 'colors.dark')};
  border-radius: 16px;
  margin: 32px;
`;
