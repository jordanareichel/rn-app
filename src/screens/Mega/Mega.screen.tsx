import React, {useEffect, useState} from 'react';
import {View} from 'react-native';

import {Container} from '@components/Container';
import {Text} from '@components/Text';
import {Header} from '@components/Header';
import {Mega as CompMega} from '@components/Mega';
import {Loader} from '@components/Loader';
import {Body, Row} from './Mega.styles';
import {Ball} from '@components/Ball';

export const Mega = () => {
  const [value, setValue] = useState(0);
  const [generateNumber, setGenerateNumber] = useState<any[]>([]);
  const [loading, setLoading] = useState(false);

  useEffect(() => {
    if (loading) {
      setTimeout(() => {
        setLoading(false);
      }, 5000);
    }
  });

  function handleGenerateNumbers() {
    setLoading(true);
    const numbers: any[] = [];
    for (let i = 0; i < value; i++) {
      numbers[i] = Math.floor(Math.random() * 60 + 1);
    }
    setGenerateNumber(numbers);
  }

  return (
    <Container
      statusBar={{
        barStyle: 'dark-content',
        backgroundColor: 'white',
      }}>
      <Header title="Mega Sena" icon="arrow-left" showNotification />
      <Loader visible={loading} text="Gerando nÃºmeros... Aguarde" size={50} />
      <Body>
        <View>
          <Text align="center" fontWeight="bold" fontSize={22}>
            Gerador de Mega Sena
          </Text>
        </View>
        {generateNumber.length > 0 && !loading && (
          <Row>
            {generateNumber.map((item, index) => (
              <Ball key={index} numbers={item} />
            ))}
          </Row>
        )}
        <CompMega
          quantity={value}
          onChangeValue={e => setValue(e)}
          onPress={() => handleGenerateNumbers()}
        />
      </Body>
    </Container>
  );
};
