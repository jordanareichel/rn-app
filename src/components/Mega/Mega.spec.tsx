import React, {useState} from 'react';

import TestRenderer from 'react-test-renderer';
import {ThemeProvider} from 'styled-components/native';
import {Mega} from './Mega.comp';
import {theme} from '@styles/theme';

const MegaComp = props => {
  const [value, setValue] = useState(0);

  return (
    <Mega
      {...props}
      quantity={value}
      onChangeValue={e => setValue(e)}
      onPress={() => console.log('')}
    />
  );
};

describe('Component Mega', () => {
  test('renderer currently', () => {
    const tree = TestRenderer.create(
      <ThemeProvider theme={theme}>
        <MegaComp />
      </ThemeProvider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
