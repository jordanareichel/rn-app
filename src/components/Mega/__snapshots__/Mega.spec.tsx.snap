// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Mega renderer currently 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "marginTop": 32,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#DEDCDB",
          "borderRadius": 4,
          "borderStyle": "solid",
          "flexDirection": "row",
          "minHeight": 56,
          "overflow": "hidden",
          "position": "relative",
          "width": "100%",
        },
      ]
    }
  >
    <TextInput
      editable={true}
      keyboardType="numeric"
      onChangeText={[Function]}
      placeholder="Digite um nÃºmero"
      style={
        Array [
          Object {
            "borderColor": "black",
            "borderStyle": "solid",
            "borderWidth": 0,
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "fontSize": 16,
            "height": "100%",
            "left": 5,
            "top": 10,
            "width": "100%",
            "zIndex": 2,
          },
        ]
      }
      value="0"
    />
  </View>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#000000",
        "borderRadius": 16,
        "marginBottom": 32,
        "marginLeft": 32,
        "marginRight": 32,
        "marginTop": 32,
        "opacity": 1,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      }
    }
  >
    <Text
      color="white"
      fontSize={16}
      fontWeight="bold"
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
            "textAlign": "center",
          },
        ]
      }
    >
      Gerar
    </Text>
  </View>
</View>
`;
